data/caasp/update.sh

```diff
@@ -92,7 +108,43 @@ elif [ ! -z "${REBOOT:-}" ]; then
 	where="-P roles:kube-(master|minion)"
 	srun="docker exec -d $saltid salt"
 	$srun $where system.reboot
+
+elif [ ! -z "${INSTALL:-}" ]; then
+
+	# Fetch all the packages included for this QAM incident
+	zypper -q --plus-content UPDATE pa -R UPDATE | awk -F'|' 'NR>2 {print $3}' > qam_packages.txt
+
+        # Check if there are any packages included into the QAM incident that are not pre-installed
+        reboot_required=0
+        while read pkg; do
+            if ! rpm -q $pkg; then
+                echo "$pkg " >> not_installed.txt
+                reboot_required=1
+            fi
+        done < qam_packages.txt
+
+        if [ ${reboot_required} -eq 1 ]; then
+            echo "The following packages are going to be installed: $(cat not_installed.txt)"
+
+            # Disable the update repo, so it will install the released version of those
+            $runner "zypper mr -d UPDATE"
+            $runner "zypper lr UPDATE | grep Enabled"
+
+            # Install the packages
+            $runner "/usr/sbin/transactional-update pkg install -y $(cat not_installed.txt)"
+
+            # Enable the update repo
+            $runner "zypper mr -e UPDATE"
+            $runner "zypper lr UPDATE | grep Enabled"
+
+	    # Reboot the cluster
+            where="-P roles:(admin|kube-(master|minion))"
```

Comment:
I am not sure how salt handles cases when you reboot salt-master together with all minions.
Can it happen that `admin` reboots before `(master|minion)` and reboot command won't be processed on all nodes?